### PR TITLE
Added hotkeys for support powers

### PR DESCRIPTION
**Pull request by sinf**
Implemented hotkey logic for support powers (as requested in #7369) and UI items for configuring the hotkeys.
Also the hotkey is displayed in the support power tooltip in the same manner as unit/structure tooltips are.

### CODE REVIEW

**Comment by Phrohdoh** [comment]
```
OpenRA.Mods.Common/Widgets/SupportPowersWidget.cs:L124: [SA1513] Statements or elements wrapped in curly brackets must be followed by a blank line.
OpenRA.Mods.Common/Widgets/SupportPowersWidget.cs:L121: [SA1008] Invalid spacing around the opening parenthesis.
OpenRA.Mods.Common/Widgets/SupportPowersWidget.cs:L121: [SA1009] Invalid spacing around the closing parenthesis.
OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs:L65: [SA1507] The code must not contain multiple blank lines in a row.
```

File: OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs [hunk 1]
```diff
@@ -64,4 +76,4 @@ public SupportPowerTooltipLogic(Widget widget, TooltipContainerWidget tooltipCon
 			descLabel.GetText = () => desc;
 		}
 	}
-}
\ No newline at end of file
+}
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs]: This hunk will need to be reverted.

File: OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs [hunk 2]
```diff
@@ -35,10 +36,13 @@ public SupportPowerTooltipLogic(Widget widget, TooltipContainerWidget tooltipCon
 			SupportPowerInstance lastPower = null;
 			tooltipContainer.BeforeRender = () =>
 			{
-				var sp = palette.TooltipPower;
-				if (sp == null)
+				var spicon = palette.TooltipIcon;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs]: The naming confused me. Maybe just call it `icon`?

**Comment by abcdefg30** [comment]
Works ingame and code looks ok to me. :+1: / :white_check_mark:

File: mods/cnc/chrome/settings.yaml [hunk 3]
```diff
@@ -451,6 +451,20 @@ Container@SETTINGS_PANEL:
 											X: PARENT_RIGHT-WIDTH+1
 											Width: 80
 											Height: 25
+								Container@SUPPORTPOWER_TEMPLATE:
```
Comment (pchote) [inline mods/cnc/chrome/settings.yaml]: These are just duplicating the production template.  Could you reuse that instead?

**Comment by pchote** [comment]
TD uses a smaller margin than RA, so offsets can't be copied directly between the two mods:

support

**Comment by pchote** [comment]
If we use shift+F1-6 for support powers then we will not be able to fix #5544.  Are people happy with this tradeoff?

**Comment by sinf** [comment]
Reused the production template thingy. Removed X coordinate from the cnc tooltip so its like the other tooltips there. Also changed the default bindings to "Unknown" so they don't conflict with anything. If a user doesn't want a feature then it shouldn't be in the way. But if a user does want to use the feature then it can easily be enabled. Isn't this the best for everyone?

**Comment by Unit158** [comment]
Why not just shift+number?

**Comment by penev92** [comment]
Because that is already used for something else.

**Comment by Unit158** [comment]
Are you sure? I didn't see it bound to anything.

**Comment by penev92** [comment]
Numbers are used for control groups. Shift is used to modify control groups. Shift + X means "add control group X to the current selection".

**Comment by Unit158** [comment]
That's ctrl+#. I'm pretty sure they're unused.

**Comment by penev92** [comment]
@pchote: I'm perfectly fine with not having that. Support power hotkeys are immensely more important.
@Unit158: I'm not going to argue over this. Go play <insert_any_RTS_game's_name_here> and see for yourself.

**Comment by abcdefg30** [comment]
> @pchote: I'm perfectly fine with not having that. Support power hotkeys are immensely more important.

Sorry, but I'm not.

**Comment by Unit158** [comment]
Shift+# is unused. Feel free to use it.

**Comment by penev92** [comment]
Seriously, I don't feel like explaining in both IRC and here. Can we drop the spam from here?
Shift + number is one of the basic key combinations for unit management in any RTS game ever, and we have it.

**Comment by Unit158** [comment]
On a US English keyboard, Shift+# does jack shit. I'll even upload a video. It is ctrl+# that creates and adds units to control groups.

**Comment by abcdefg30** [comment]
Let's stay with what @sinf suggested (unkown keycode).

**Comment by Mailaender** [comment]
:+1: Works as promised. Thanks!

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/acf34bd1c87e7cc52f99778f83413c4c89439ed9...7e447a50da156681ac3b4910a5fcccd546978fc4)

**Pull request merged**